rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+> [[Source]](https://semantic-release.gitbook.io/semantic-release/support/faq#how-can-i-revert-a-release)
+
+## Release Notes
+
+In version 4.3.0 we introduced [`renogen`](https://github.com/DDAZZA/renogen) a ruby gem to allow us to write release notes as a collection of changes. Prior
+to this I believe they were written and updated by each collaborator.
+
+[`semantic-release`](https://github.com/semantic-release/semantic-release) has a package called [`@semantic-release/changelog`](https://github.com/semantic-release/changelog) when used in conjunction with [`@semantic-release/git`](https://github.com/semantic-release/git) and
+[`@semantic-release/npm`](https://github.com/semantic-release/npm) it will maintain a changelog that is committed to git and published to [NPM](https://www.npmjs.com/). We will use this to help
+communicate the features that each release brings. The changelog will include the `TYPE`, `SCOPE` and `DESCRIPTION`, it
+will not include the `BODY` but it will include a link to the relevant pull request.
+
+## Maintenance Releases
+The beta version of [`semantic-release`](https://github.com/semantic-release/semantic-release) includes extensive support for publishing a maintenance
+release. There is a [detailed walk through](https://github.com/semantic-release/semantic-release/blob/916c2685c57f3490fb1e50afbf72ea8dce11e188/docs/recipes/maintenance-releases.md)
+outlining many scenarios.
+
+This functionality is not available in the latest stable release, however the beta has been evolving for in excess of a year.
+
+
+# Drawbacks
+- There is a learning curve to this new process
+- Version numbers will increase substantially if we don't plan our interfaces carefully to prevent breaking changes. I
+don't consider this to be a drawback, but we need to be aware about breaking changes. Previously we have bundled many
+breaking changes into one release.
+- Changelog format will change
+
+# Alternatives
+There are alternatives to [`semantic-release`](https://github.com/semantic-release/semantic-release) for automating the
+release process such as [np](https://github.com/sindresorhus/np)
+
+However, I could find no other solution that parses commits to work out the version number. This means that someone has
+to be involved in the release process to determine the version number. This is not sustainable and prone to errors.
+
+# Adoption strategy
+
+This RFC PR includes all changes necessary, if this RFC is accepted when it is merged it will immediately start
+releasing from the following branches `['+([0-9])?(.{+([0-9]),x}).x', 'master', 'next', 'next-major', {name: 'beta', prerelease: true}, {name: 'alpha', prerelease: true}]`  
+[[Source]](https://github.com/semantic-release/semantic-release/blob/916c2685c57f3490fb1e50afbf72ea8dce11e188/docs/usage/configuration.md#branches)
```

Comment:
I think we should also link to this documentation:

* https://github.com/semantic-release/semantic-release/blob/v16.0.0-beta.43/docs/usage/workflow-configuration.md#branches-properties

Reply:
added in 2d883599e